### PR TITLE
Added logic to download and extract docker from download.docker.com. …

### LINKED ISSUE #256: install-docker-ce.ps1 defaults to nightly builds
**Describe the bug**
`install-docker-ce.ps1` which is mentioned in [this](https://docs.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=dockerce#windows-server-1) documentation defaults to non-released, nightly builds of Docker:
https://github.com/microsoft/Windows-Containers/blob/78519ddbc455916f677aead0aa91f5711c91f25d/helpful_tools/Install-DockerCE/install-docker-ce.ps1#L65-L71
which is very bad idea people will end up to seeing weird issues on their environments because of that. Proper way would be default to latest stable binaries which can be found from https://download.docker.com/win/static/stable/x86_64/

**To Reproduce**
Install like it is described in documentation.

**Expected behavior**
That script default to latest stable version of Docker.

**Configuration:**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by brasmith-ms**
…Also updated the readme. Fixes #256

### CODE REVIEW

File: helpful_tools/Install-DockerCE/README.md [hunk 1]
```diff
@@ -1,47 +1,66 @@
-## Install-ContainerHost.ps1
+## install-docker-ce.ps1
 
 #### NAME
-    Install-ContainerHost.ps1
+    install-docker-ce.ps1
     
 #### SYNOPSIS
-    Installs the prerequisites for creating Windows containers
+    Installs the prerequisites for running Windows containers with docker CE
```
Comment (neersighted) [inline helpful_tools/Install-DockerCE/README.md]:
```suggestion
    Installs the prerequisites for running Windows containers with Docker CE
```

File: helpful_tools/Install-DockerCE/install-docker-ce.ps1 [hunk 2]
```diff
@@ -46,7 +49,9 @@
         If a restart is required the script will terminate and will not reboot the machine
 
     .PARAMETER ContainerBaseImage
-        Use this to specifiy the URI of the container base image you wish to pull
+        Use this to specify the URI of the container base image you wish to pull
```
Comment (neersighted) [inline helpful_tools/Install-DockerCE/install-docker-ce.ps1]:
```suggestion
        Use this to specify the URI of the container base image you wish to pre-pull
```

File: helpful_tools/Install-DockerCE/README.md [hunk 3]
```diff
@@ -60,30 +79,31 @@
...
         Required?                    false
         Position?                    named
         Default value                False
         Accept pipeline input?       false
         Accept wildcard characters?  false
         
-    -SkipImageImport [<SwitchParameter>]
-        Skips import of the base WindowsServerCore image.
+    -ContainerBaseImage [<String>]
+        Use this to specify the URI of the container base image you wish to pull
```
Comment (neersighted) [inline helpful_tools/Install-DockerCE/README.md]:
```suggestion
        Use this to specify the URI of the container base image you wish to pre-pull
```

File: helpful_tools/Install-DockerCE/install-docker-ce.ps1 [hunk 4]
```diff
@@ -571,10 +576,49 @@ Install-Docker()
...
+        Write-Output "DOCKER $DockerPath"
+        if ($DockerPath -eq "default" -or $DockerDPath -eq "default") {
+            Write-Output "Checking Docker version"
+            $version = ((Invoke-WebRequest -Uri $DefaultDockerLocation).Links | Where-Object {$_.href -like "docker*"}).href | Sort-Object -Descending | Select-Object -First 1  | Select-String -Pattern "docker-(\d+\.\d+\.\d+).+"  -AllMatches | Select-Object -Expand Matches | %{ $_.Groups[1].Value }
+
+            if($DockerVersion -ne "latest") {
+                $version = $DockerVersion
+            } else {
+                if(!(((Invoke-WebRequest -Uri $global:DefaultDockerLocation).Links | Where-Object {$_.href -like "*$global:DockerVersion*"}).href | Sort-Object -Descending | Select-Object -First 1)) {
+                    Write-Error "Docker version supplied $global:DockerVersion was invalid, using the latest version."
```
Comment (neersighted) [inline helpful_tools/Install-DockerCE/install-docker-ce.ps1]: It feels like these line should be less dense/not one-liners, using multiple assignments to increase readability. Falling back to latest when getting an invalid version also seems somewhat unexpected.

**Review by neersighted — COMMENTED** [review]
Looks overall quite reasonable to me, cc @cpuguy83 as I know you have some interest in this topic, and since this script is used by the GitHub Actions team and not just users reading the Windows Containers docs.

**Review by akarshm — DISMISSED** [review]

File: helpful_tools/Install-DockerCE/install-docker-ce.ps1 (hunk 4 above)
Comment (brasmith-ms) [reply helpful_tools/Install-DockerCE/install-docker-ce.ps1]: Good point, I just updated the logic. Any better?

**Review by neersighted — APPROVED** [review]
LGTM with the caveat that I'm not a PowerShell expert and just daytrip -- but from a design standpoint, this looks quite solid to me, and makes this script feel like something that can be advertised to users in the Microsoft docs.

**Review by akarshm — APPROVED** [review]

**Pull request merged**